tls/s2n_client_hello.c

```diff
@@ -172,13 +172,16 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
     GUARD(s2n_stuffer_erase_and_read_bytes(in, conn->secure.client_random, S2N_TLS_RANDOM_DATA_LEN));
     GUARD(s2n_stuffer_read_uint8(in, &conn->session_id_len));
 
-    conn->client_protocol_version = (client_protocol_version[0] * 10) + client_protocol_version[1];
+    conn->client_protocol_version = MIN((client_protocol_version[0] * 10) + client_protocol_version[1], S2N_TLS12);
     conn->client_hello_version = conn->client_protocol_version;
     /* Protocol version in the ClientHello is fixed at 0x0303(TLS 1.2) for
      * future versions of TLS. Still, we will negotiate down if a client sends
      * an unexpected value above 0x0303.
      */
-    conn->actual_protocol_version = MIN(conn->client_protocol_version, conn->server_protocol_version);
+    if (conn->server_protocol_version < S2N_TLS13)
```

Comment:
It looks like this change didn't break any tests (so we must not have a test for this). Are you able to add a test that sending 0x0304 in the client protocol doesn't result in a tls1.3 negotiation?

Reply:
good point, will do